draft-romann-mud-constrained.md

```diff
@@ -385,36 +398,44 @@ For the discovery process described in {{general_discovery}}, the following cons
 
 - MUD receivers MUST regularly perform a CoAP request to the "All MUD CoAP Nodes" multicast address for the `/.well-known/mud-url` URI.
```

Comment:
Not related to this PR, but should this really be a MUST? And should we differentiate the IPv4/IPv6 case here as well?

Reply:
1. My idea here was that by requiring the multicast request, we have at least one method of discovery that Things can assume to always be working. However, on second thought, this might be problematic if the underlying network does not allow for multicast, so maybe we should either specify the situations in which this is required, or replace MUST with SHOULD. What are your thoughts on this?

2. Probably, yes, for clarity. I'll add that.

Reply:
> My idea here was that by requiring the multicast request, we have at least one method of discovery that Things can assume to always be working. However, on second thought, this might be problematic if the underlying network does not allow for multicast, so maybe we should either specify the situations in which this is required, or replace MUST with SHOULD. What are your thoughts on this?

I was also thinking about changing MUST to SHOULD here. Maybe we could add a MUST assertion, though, that demands that a MUD receiver has to at least support one discovery method? Or should it also be possible that a MUD receiver only supports MUD URL submission?

Reply:
Depending on the environment, it might be desirable to have MUD receivers that only support submission, so I think just replacing MUST with SHOULD is sufficient here.